master/internal/stream/system_intg_test.go

```diff
@@ -349,3 +349,273 @@ func TestTrialUpdate(t *testing.T) {
 
 	require.NoError(t, errgrp.Wait())
 }
+
+func TestExperimentStartup(t *testing.T) {
+	// setup test environment
+	superCtx, ctx, testUser, ps, socket, pgDB, dbCleanup := setupStreamTest(t)
+	t.Cleanup(dbCleanup)
+
+	testCases := []startupTestCase{
+		{
+			description: "experiment subscription with experiment id",
+			startupMsg: StartupMsg{
+				SyncID: "1",
+				Known: KnownKeySet{
+					Experiments: "1",
+				},
+				Subscribe: SubscriptionSpecSet{
+					Experiments: &ExperimentSubscriptionSpec{
+						ExperimentIds: []int{1},
+						Since:         0,
+					},
+				},
+			},
+			expectedSync:      "key: sync_msg, sync_id: 1",
+			expectedUpserts:   []string{},
+			expectedDeletions: []string{"key: experiments_deleted, deleted: "},
+		},
+		{
+			description: "experiment subscription with extra known experiments",
+			startupMsg: StartupMsg{
+				SyncID: "2",
+				Known: KnownKeySet{
+					Experiments: "1,3,4", // 3 and 4 do not exist
+				},
+				Subscribe: SubscriptionSpecSet{
+					Experiments: &ExperimentSubscriptionSpec{
+						ExperimentIds: []int{1},
+						Since:         0,
+					},
+				},
+			},
+			expectedSync:      "key: sync_msg, sync_id: 2",
+			expectedUpserts:   []string{},
+			expectedDeletions: []string{"key: experiments_deleted, deleted: 3-4"},
+		},
+		{
+			description: "experiment subscription with incomplete known experiments",
+			startupMsg: StartupMsg{
+				SyncID: "3",
+				Known: KnownKeySet{
+					Experiments: "1,4", // 2 is not known, and 4 does not exist
+				},
+				Subscribe: SubscriptionSpecSet{
+					Experiments: &ExperimentSubscriptionSpec{
+						ExperimentIds: []int{1, 2, 3, 4},
+						Since:         0,
+					},
+				},
+			},
+			expectedSync:      "key: sync_msg, sync_id: 3",
+			expectedUpserts:   []string{"key: experiment, exp_id: 2, state: ERROR, project_id: 2, job_id: test_job2"},
+			expectedDeletions: []string{"key: experiments_deleted, deleted: 4"},
+		},
+		{
+			description: "experiment subscription with divergent known set",
+			startupMsg: StartupMsg{
+				SyncID: "4",
+				Known: KnownKeySet{
+					Experiments: "1",
+				},
+				Subscribe: SubscriptionSpecSet{
+					Experiments: &ExperimentSubscriptionSpec{
+						ExperimentIds: []int{2},
+					},
+				},
+			},
+			expectedSync:      "key: sync_msg, sync_id: 4",
+			expectedUpserts:   []string{"key: experiment, exp_id: 2, state: ERROR, project_id: 2, job_id: test_job2"},
+			expectedDeletions: []string{"key: experiments_deleted, deleted: 1"},
+		},
+	}
+
+	errgrp := errgroupx.WithContext(ctx)
+	trials := streamdata.GenerateStreamTrials()
```

Comment:
question: Do we want to have a separate SQL file to run for setting up experiments? Or changing the name of this function to make it clear that it is doing more than just setting up the DB for streaming trials?